===== Patch instructions generated by SAVER =====
[+] { Insert tmp := do_setup_cgroup_limits:iterator->next at 4 (line 1925, column 2) and replace the original expression at 6 (line 1925, column 2) under true }
===== Actual patch in diff format =====
@@ -1922,7 +1922,8 @@ static int do_setup_cgroup_limits(struct cgfs_data *d,
 	ret = 0;
 	INFO("cgroup has been setup");
 out:
-	lxc_list_for_each(iterator, sorted_cgroup_settings) {
+	for(iterator = sorted_cgroup_setting->next; iterator != sorted_cgroup_setting; iterator = tmp) {
+		tmp = iterator->next;
 		lxc_list_del(iterator);
 		free(iterator);
 	}
